backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopAnalysisCallback.scala

```diff
@@ -354,7 +354,14 @@ final class BloopAnalysisCallback(
     }
   }
 
-  override def apiPhaseCompleted(): Unit = ()
+  override def apiPhaseCompleted(): Unit = {
+    // We collect all the generated classes during `API`. When this phase completes, we can pass to
+    // the classfiles manager all the classes that were generated by scalac. This will mark them as
+    // invalidated, so that javac (if it runs after) can be notified.
+    // NOTE: When the compilation order is `JavaThenScala`, scalac will be notified about the
```

Comment:
Hmm, I don't think this is true. `CallbackGlobal` does not call `generated` and none of the generation logic depends on the compiler order. What did you mean?

Reply:
I meant that when we start by compiling the Java sources, we'll need to tell scalac which classfiles have been invalidated by this compilation step. This information is passed to scalac from `CallbackGlobal`.